docs/sphinx/user_guide/feature/view.rst

```diff
@@ -330,6 +330,55 @@ indices by :math:`N`, ::
     }
   }
 
+Index Layout
+^^^^^^^^^^^^^  
+``RAJA::IndexLayout`` is a layout that uses an index list to access entries of a 
+view.  This layout requires an indexing strategy for each dimension, which 
+determines how to map a given input index to an entry within a view.
+
+Three indexing strategies are natively supported in RAJA:
+DirectIndex, IndexList, and ConditionalIndexList. DirectIndex maps an input
+index to itself, and does not take any arguments as its constructor.  The
+IndexList strategy takes a reference to an array containing the list of relevant
+indices to map.  With this strategy, a given input index is mapped to the entry
+in its list corresponding to that index.  Lastly, the ConditionalIndexStrategy
+takes a reference to an array containing an index list. When the referenced index
+list is non-null, the ConditionalIndex strategy is equivalent to that
+of the IndexList.  Otherwise, if the index list provided to the constructor is a
+null pointer, the ConditionalIndexList maps an input index to itself (same as
+the DirectIndex strategy).  The ConditionalIndexList strategy is useful if the
+index list can potentially be unused for some situations.
+
+A simple illustrative example is shown below::
+
+  int data[2][3];
+
+  for (int i = 0; i < 2; i ++ ) {
+    for (int j = 0; j < 3; j ++ ) {
+      // fill data[i][j]...
+    }
+  }
+
+  int index_list[2] = {1,2};
+
+  auto index_tuple = camp::tuple<RAJA::DirectIndex<>, RAJA::IndexList<>>(

```

Comment:
```suggestion
  auto index_tuple = RAJA::tuple<RAJA::DirectIndex<>, RAJA::IndexList<>>(

```

We have a file `RAJA/include/RAJA/util/camp_aliases.hpp` that provides aliases to camp types in the RAJA namespace. This is preferred to make it simpler for users (they don't have to remember whether a type is in RAJA or camp).